.github/workflows/workflow-build-otelcol-config.yml

```diff
@@ -0,0 +1,166 @@
+
+      - uses: actions/cache/restore@v4
+        with:
+          path: |
+            ${{ env.GOMODCACHE }}/cache
+            ${{ env.GOCACHE }}
+          key: ${{ steps.get-cache-key.outputs.cache-key }}
+          restore-keys: |
+            ${{ steps.get-cache-key.outputs.restore-keys }}
+
+      - name: Build
+        if: '! inputs.fips'
+        run: make otelcol-config-${{inputs.arch_os}}
+        working-directory: ./pkg/tools/otelcol-config
+
+      - uses: actions/cache/restore@v4
+        id: restore-toolchain-cache
+        if: inputs.fips && contains(inputs.arch_os, 'linux')
+        with:
+          path: |
+            /opt/toolchain
+          key: ${{ steps.get-cache-key.outputs.toolchain-cache-key }}
+
+      # TODO: pass go version to toolchain make target
+      - name: Rebuild Toolchains
+        id: rebuild-toolchain
+        if: ${{ steps.restore-toolchain-cache.outcome == 'success' && steps.restore-toolchain-cache.outputs.cache-hit != 'true' }}
+        run: make toolchain-${{ inputs.arch_os }} OUTPUT=/opt/toolchain -j3
+        working-directory: ./toolchains
+
+      - name: Build (FIPS)
+        if: inputs.fips && contains(inputs.arch_os, 'linux')
+        run: |
+          CC=$(find /opt/toolchain/bin -type f -name "*-linux-musl-gcc")
+          test "$CC"
+          echo "Using toolchain: $CC"
+          make otelcol-config-${{inputs.arch_os}} \
+            FIPS_SUFFIX="-fips" \
+            CGO_ENABLED="1" \
+            CC="$CC" \
+            EXTRA_LDFLAGS="-linkmode external -extldflags '-static'"
+        working-directory: ./pkg/tools/otelcol-config
+
+      - name: Build (FIPS)
+        if: inputs.fips && contains(inputs.arch_os, 'linux')
+        run: |
+          CC=$(find /opt/toolchain/bin -type f -name "*-linux-musl-gcc")
+          test "$CC"
+          echo "Using toolchain: $CC"
+          make otelcol-config-${{inputs.arch_os}} \
+            FIPS_SUFFIX="-fips" \
+            CGO_ENABLED="1" \
+            CC="$CC" \
+            EXTRA_LDFLAGS="-linkmode external -extldflags '-static'"
+        working-directory: ./pkg/tools/otelcol-config
+
+      - name: Set binary name
+        id: set-binary-name
+        run: echo "binary_name=otelcol-config${FIPS_SUFFIX}-${{inputs.arch_os}}" >> $GITHUB_OUTPUT
+
+      - name: Show BoringSSL symbols
+        if: inputs.fips && contains(inputs.arch_os, 'linux')
+        working-directory: ./pkg/tools/otelcol-config
+        run: |
+          go tool nm ${{ steps.set-binary-name.outputs.binary_name }} | \
+          grep "_Cfunc__goboringcrypto_"
+
+      # TODO: find a way to test if the binary runs
+      # - name: Test binary
+      #   if: inputs.arch_os == env.ARCH_OS
+      #   working-directory: ./pkg/tools/otelcol-config
+      #   run: |
+      #     ./${{ steps.set-binary-name.outputs.binary_name }} -t foo
```

Comment:
There's already a smoke test in otelcol-config itself, so this is probably not needed

Reply:
This was copied from the otelcol-sumo build job. It's meant to ensure that the binary runs after running make. We ran into an issue where the binaries were actually text files in the past.

It may be better to check the file type for every platform instead. We'd need to check for the correct file type for the platform though (e.g. `Mach-O 64-bit executable arm64` versus `ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, Go BuildID=vataBbMZrfQgeqsJPb3y/CT0CNbBkuU-ZpeiBCFvZ/rX1IRAQtPBnQNJWqqDt9/SENzp0KECm910ny3Zm29, stripped`)

Reply:
That seems kind of nitty to me. How did it regress to a text file?

Reply:
I'm not positive but I believe it was a problem with binaries moving between jobs through the CI cache system which isn't happening the same way here. We can leave this out.